### DISCUSSION TITLE
TypeError: [E1046] KnowledgeBase is an abstract class and cannot be instantiated

### DISCUSSION THREAD

**Discussion by ghost** [explosion/spaCy#13703] [Help: Coding & Implementations]
During the poetry train, i am getting the below error. Can you please help me how to resolve the issue ?
 kb = KnowledgeBase(vocab=nlp.vocab, entity_vector_length=vectors_dim)
  File "spacy/kb/kb.pyx", line 27, in spacy.kb.kb.KnowledgeBase.__init__
TypeError: [E1046] KnowledgeBase is an abstract class and cannot be instantiated. If you are looking for spaCy's default knowledge base, use `InMemoryLookupKB`.
Error: Sequence aborted after failed subtask 'train[2]'

**Comment by DuyguA** [comment]
Yes, I think classes have changed a bit. I do this with the latest version:

```
from spacy.kb import InMemoryLookupKB
kb = InMemoryLookupKB(vocab=nlp.vocab, entity_vector_length=nlp.vocab.vectors_length)

for qid, desc in descriptions.items():
    desc_doc = nlp(desc)
    desc_enc = desc_doc.vector
    kb.add_entity(entity=qid, entity_vector=desc_enc, freq=20) # freqs are uniform and random
    kb.add_alias(alias=alias, entities=[qid], probabilities=[1.0])

kb.to_disk("my_kb")
```
